### ISSUE TITLE
Cannot create content-type named "Subscription" (Graphql conflict)

### ISSUE THREAD

**Issue by 5ika** [strapi/strapi#18638]
### Required System information

- Node.js version: v18.18.2
- Yarn version: 4.0.0
- Strapi version: v4.15.0
- Database: Postgres
- Operating system: Ubuntu 23.04
- Is your project Javascript or Typescript: Typescript

### Describe the bug
I created a content-type named "Subscription" and add public permissions for `find` and `findOne`. When I try to query with Graphql my new model, I have a `Forbidden access` error but only when I try to get `attributes` (see below).

My Graphql query:
```gql
query subscriptions {
  subscriptions {
    data {
      id
      attributes {
        name 
      }
    }
  }
}
```

Result
```json
{
  "errors": [
    {
      "message": "Forbidden access",
      "extensions": {
        "error": {
          "name": "ForbiddenError",
          "message": "Forbidden access",
          "details": {}
        },
        "code": "FORBIDDEN"
      }
    }
  ],
  "data": {
    "subscriptions": {
      "data": [
        {
          "id": "1",
          "attributes": null
        }
      ]
    }
  }
}
```

After two hours of investigation, I find this line in Graphql plugin: https://github.com/strapi/strapi/blob/4ef442cc491bc4a51f10574f30f5b368fcfd1b2a/packages/plugins/graphql/server/services/content-api/wrap-resolvers.js#L114

At some point, the content-type name is passed as value of `type` and `isValidType` is `true`  because there is a conflict between my content-type name and the *Subscription* general method of GraphQL.

### Steps to reproduce the behavior

1. Create a content-type named "Subscription"
2. Give public permission on this model for `find` and `findOne`
3. Try to list subscriptions with a Graphql request

### Expected behavior

I can list subscriptions with Graphql without permission errors OR I can't create a model named `Subscription`.

### Screenshots

![image](https://github.com/strapi/strapi/assets/7430173/ebaaf687-5e5b-4a62-9681-f384ae002407)

**Comment by nick-aj** [comment]
Hello @5ika,

Thank you very much for submitting this bug report to us, unfortunately I was not fully able to reproduce.
_Here are some of my notes though:_

- When I tried with find and findOne disabled for the Public role, I did get the same issue that you report.
- However, after marking find and findOne for Public role, I am able to get the data just right:
<img width="1068" alt="image" src="https://github.com/strapi/strapi/assets/145370380/6cf12d3a-8d21-4946-b12c-1141e9a0cc3e">

NOTE: I tested using Strapi version 4.15.1, make sure you use the latest version to leverage all available bug fixes.
